src/PineconeClient.cs

```diff
@@ -38,58 +34,85 @@ public PineconeClient(string apiKey, HttpClient client)
         Http.DefaultRequestHeaders.Add("Api-Key", apiKey);
     }
 
-    public async Task<string[]> ListIndexes()
+    public async Task<IndexDetails[]> ListIndexes(CancellationToken cancellationToken = default)
     {
-        var indexes = await Http
-            .GetFromJsonAsync("/databases", SerializerContext.Default.StringArray)
+        var listIndexesResult = (ListIndexesResult?)await Http
+            .GetFromJsonAsync("/indexes", typeof(ListIndexesResult), SerializerContext.Default, cancellationToken)
             .ConfigureAwait(false);
 
-        return indexes ?? [];
+        return listIndexesResult?.Indexes ?? [];
     }
 
-    public Task CreateIndex(string name, uint dimension, Metric metric) =>
-        CreateIndex(new IndexDetails { Name = name, Dimension = dimension, Metric = metric });
+    public Task CreatePodBasedIndex(string name, uint dimension, Metric metric, string environment, string podType, long pods, CancellationToken cancellationToken = default)
```

Comment:
Can the users now choose the environment of pod-based indexes with free accounts? (if pod-based indexes are still a thing for those)

It was mostly a shorthand overload given in most situations only those three arguments were needed. It's not so shorthand anymore so you may want to either add sensible defaults to arguments (or omit them so the create request leaves them unspecified if allowed) or just remove it altogether if making it work is unfeasible/unsound.

Reply:
For "gcp-starter" it doesn't matter what pod type, number of pods, etc. you specify. As for the environment - anything other than "gcp-starter" will fail on free account. Sensible defaults (like "p1.x1") seem to be a good idea and should be easy to make it work across the board. However, my preference would be to leave the environment variable without the default, rather than default to "gcp-starter" - for better discoverability. Was planning to add xml comments in a bit, so could add a note about "gcp-starter" for a free user. What do you think?